Dockerfile

```diff
@@ -0,0 +1,27 @@
+# Use Alpine-node as a base image because it is much smaller.
+#
+# Base image size for mhart/alpine-node:4 is 13 MB
+# Base image for standard node:4 is 253 MB.
+#
+# Our production deploy does not have any C/C++ library dependencies yet
+# so this is still easy to do.
+FROM mhart/alpine-node:4
```

Comment:
Wow you're thorough. :)  paxctl is command that modifies the binary to set some flags that are examined if your kernel is built with grsecurity hardening.  grsecurity is a cool suite of tools that allow for much stricter memory protections. For example, you can mark your stack and heap as non-executable completely eliminating the standard buffer-overflow vulnerability for code injection.  However, node.js does JIT so it needs to be able to write executable code. This command is marking the node executable as needing to disable those memory protections.

As for why this image was chosen, I originally thought it'd be best to use CentOS or something...but my understanding has changed recently.  Eric Maland pointed out that docker is conceptually one task. The fact that the container has to have an OS to bootstrap that process is sorta incidental. IIRC, the ideal docker setup is actually a 100% statically linked binary with almost no supporting filesystem.

Thus, ansible has no place in docker. Instead, you would create one docker image per functional unit (making up a term here) and then use docker-compose to weave them together into what would be the equivalent of a playbook.

In this world, all standard distros are too heavy weight. Alpine linux is a super stripped down linux that's meant to just run one binary. Thus using alpine-node (or alpine based nginx) makes the most sense to me for a container OS....